omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/hiv/ProgramLocationWarningFragmentController.java

```diff
@@ -33,7 +33,12 @@ public void controller(FragmentModel model,
             if (patient !=null && hivProgram !=null) {
                 List<PatientProgram> patientPrograms = programWorkflowService.getPatientPrograms(patient, hivProgram, null, null, null, null, false);
                 if (patientPrograms != null && patientPrograms.size() > 0 ) {
-                    model.addAttribute("hivProgramLocation", patientPrograms.get(0).getLocation());
+                    if(patientPrograms.get(0).getLocation() !=null){
+                      model.addAttribute("hivProgramLocation", patientPrograms.get(0).getLocation());
+                    }else{
+                     model.addAttribute("error", "No Location Found for this patient");
```

Comment:
Hi @Dutervil , this looks OK. Another way to safeguard agains program location null is to set the attribute up to the top:
` model.addAttribute("sessionLocation", uiSessionContext.getSessionLocation());
+        model.addAttribute("hivProgramLocation", null);`
and then in the gsp page you check if the location is null and write the same error.
It's basically the same thing but avoids the need to introduce another model parameter, 'error'. 
Please let me know if this makes sense.